source/solvers/tracer.cc

```diff
@@ -530,6 +553,179 @@ Tracer<dim>::calculate_tracer_statistics()
+
+  // Initialize tracer information
+  std::vector<double>         tracer_values(n_q_points_face);
+  std::vector<Tensor<1, dim>> tracer_gradient(n_q_points_face);
+  FEFaceValues<dim>           fe_face_values_tracer(*this->mapping,
+                                          this->dof_handler.get_fe(),
+                                          *this->face_quadrature,
+                                          update_values | update_gradients |
+                                            update_JxW_values |
+                                            update_normal_vectors);
+
+  // Initialize fluid dynamics information
+  const DoFHandler<dim> *dof_handler_fd =
+    multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
+  const FEValuesExtractors::Vector velocities(0);
+  std::vector<Tensor<1, dim>>      velocity_values(n_q_points_face);
+  FEFaceValues<dim>                fe_face_values_fd(*this->mapping,
+                                      dof_handler_fd->get_fe(),
+                                      *this->face_quadrature,
+                                      update_values);
+
+  // Initialize fluid properties
+  auto &properties_manager =
+    this->simulation_parameters.physical_properties_manager;
+  std::map<field, std::vector<double>> fields;
+
+  const auto diffusivity_model = properties_manager.get_tracer_diffusivity();
+
+  std::vector<double>     tracer_diffusivity(n_q_points_face);
+  std::vector<Point<dim>> face_quadrature_points;
+
+  std::vector<double> tracer_flow_rate_vector(
+    this->simulation_parameters.boundary_conditions.size, 0);
+
+  // Get vector of all boundary conditions
+  const auto boundary_conditions_ids =
+    this->simulation_parameters.boundary_conditions.id;
+
+  for (const auto &cell : this->dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned() && cell->at_boundary())
+        {
+          for (const auto face : cell->face_indices())
+            {
+              if (cell->face(face)->at_boundary())
+                {
+                  const auto boundary_id =
+                    std::find(begin(boundary_conditions_ids),
+                              end(boundary_conditions_ids),
+                              cell->face(face)->boundary_id());
+
+
+                  unsigned int vector_index =
+                    boundary_id - boundary_conditions_ids.begin();
+
+                  // Gather tracer information
+                  fe_face_values_tracer.reinit(cell, face);
+                  fe_face_values_tracer.get_function_values(
+                    this->present_solution, tracer_values);
+                  fe_face_values_tracer.get_function_gradients(
+                    this->present_solution, tracer_gradient);
+
+                  // We update the fields required by the diffusivity
+                  // model
+                  fields.clear();
+                  if (diffusivity_model->depends_on(field::levelset))
```

Comment:
Is there a case where fields don't depend on levelset?

Reply:
Yes there can be, for example when we use the tracer without an immersed solid

Reply:
Yes that's it, since it's a map there is simply no entry